### PR TITLE
Enable construction and queries for empty trees

### PR DESCRIPTION
**Pull request by JoshChristie**
addresses issue #35

### CODE REVIEW

File: src/knn.jl [hunk 1]
```diff
@@ -1,4 +1,11 @@
-check_k(tree, k) = (k > length(tree.data)|| k <= 0) && throw(ArgumentError("k > number of points in tree or ≦ 0"))
+function check_k(tree, k)
+    if isempty(tree.data)
+        k = 0
```
Comment (KristofferC) [inline src/knn.jl]: This seems like a weird special case to me. Either k should be allowed to be bigger than the number of points and in that case we truncate it to be equal to the number of points or we error.

Comment (JoshChristie) [inline src/knn.jl]: Okay, updated it to truncate when bigger than number of points.
```julia
if k > length(tree.data)
    k = length(tree.data)
elseif k < 0
    throw(ArgumentError("k > number of points in tree or ≦ 0"))
end
```

File: src/knn.jl [hunk 2]
```diff
@@ -1,4 +1,8 @@
-check_k(tree, k) = (k > length(tree.data)|| k <= 0) && throw(ArgumentError("k > number of points in tree or ≦ 0"))
+function check_k(tree, k)
+    if k > length(tree.data) || k <= 0
```
Comment (JoshChristie) [inline src/knn.jl]: Yes it should

File: src/knn.jl [hunk 3]
```diff
@@ -1,4 +1,8 @@
-check_k(tree, k) = (k > length(tree.data)|| k <= 0) && throw(ArgumentError("k > number of points in tree or ≦ 0"))
+function check_k(tree, k)
+    if k > length(tree.data) || k < 0
+        throw(ArgumentError("k > number of points in tree or ≦ 0"))
```
Comment (KristofferC) [inline src/knn.jl]: need to update this message to be `< 0`.

**Comment by JoshChristie** [comment]
I think this is ready to go. Thanks for all the feedback.

**Comment by KristofferC** [comment]
Sorry, I merged another PR which made a conflict with this one. If you rebase this one I will merge it :)

**Comment by JoshChristie** [comment]
Ok rebased :)

**Pull request merged**